plugins/kubernetes/app/models/kubernetes/release_doc.rb

```diff
@@ -77,13 +77,46 @@ def resource_template=(value)
 
     # dynamically fill out the templates and store the result
     def store_resource_template
+      add_pod_disruption_budget
       counter = Hash.new(-1)
       self.resource_template = raw_template.map do |resource|
         index = (counter[resource.fetch(:kind)] += 1)
         TemplateFiller.new(self, resource, index: index).to_hash
       end
     end
 
+    def add_pod_disruption_budget
+      min_available = nil
+      raw_template.detect do |t|
+        min_available ||= t.dig(:metadata, :annotations, :"samson/minAvailable")
+      end
+      return unless min_available
+
+      target = if percent = min_available.to_s[/\A(\d+)\s*%\z/, 1]
```

Comment:
When there's an inline regex like this, useful to have a comment explaining what the regex is supposed to be doing.